src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -347,19 +345,24 @@
    [:ShutdownService shutdown-on-error]]
 
   (start [this context]
-         (start-puppetdb context (get-config) this add-ring-handler get-route shutdown-on-error))
-
-  (stop [this context]
-        (stop-puppetdb context false))
-  (shared-globals [this]
-                  (:shared-globals (service-context this)))
+         (let [config (-> (get-config)
+                          conf/process-config!)
+               url-prefix (get-route this)
+               shutdown-on-error* (partial shutdown-on-error (service-id this))
+               add-ring-handler* #(->> %
+                                       (compojure/context url-prefix [])
+                                       (add-ring-handler this))]
```

Comment:
similar question here -- is this just moving function calls out of start-puppetdb and into puppetdb-service or is there something else going on?